### PR TITLE
Add Android lint & test to GitHub CI

### LINKED ISSUE #6678: Run Android tests on Github Actions
When /app-android path is changed we should run Android tests.
We should ideally also require them to pass.

### PR DESCRIPTION
**Pull request by murilopereirame**
This commit implements the Android Lint and Tests into the pipeline to check if everything is ok at least with the written unit tests.

Closes #6678

### CODE REVIEW

File: .github/workflows/kotlin-test.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    paths: [ "app-android/**" ]
+
+jobs:
+  test-kotlin:
+    runs-on: ubuntu-latest
+    env:
+      java-version: 21
+      java-distribution: 'temurin'
+      node-version: 20.11.1
+      TZ: "Europe/Berlin"
```
Comment (charlag) [inline .github/workflows/kotlin-test.yml:16]: we should comment why we put it there and that we should remove it eventually

File: .github/workflows/kotlin-test.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        run: |
+          echo "better_sqlite3_path=$(node buildSrc/getNativeCacheLocation.js better-sqlite3)" >> $GITHUB_ENV
+      - name: Try to use cached better-sqlite3
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.better_sqlite3_path }}
+          key: ${{ env.better_sqlite3_path }}
+      - name: Install packages
+        run: npm ci && npm run build-packages
+      - name: Build WebApp
```
Comment (charlag) [inline .github/workflows/kotlin-test.yml:50]: do you really need to build the webapp to run tests? I don't think so tbh

Comment (murilopereirame) [reply .github/workflows/kotlin-test.yml:50]: Unfortunately yes, it tries to find the webapp inside the assets and if it doesn't, it creates a file pointing to app's build folder and take the files from there

Comment (charlag) [reply .github/workflows/kotlin-test.yml:50]: I deleted `build` folder for the webclient and then successfully run `./gradlew test`. Am I missing something?

**Review by charlag — APPROVED** [review]

File: .github/workflows/kotlin-test.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: Kotlin CI
+
+on:
+  push:
+    branches-ignore:
+      - master
+    paths: [ "app-android/**" ]
```
Comment (rezbyte) [inline .github/workflows/kotlin-test.yml:7]:
```suggestion
        paths:
      - 'app-android/**'
```
I have discovered square brackets in YAML are not arrays and are not recognized as such by GitHub Actions. This is the style used by the GitHub Actions' documentation.

Comment (charlag) [reply .github/workflows/kotlin-test.yml:7]: both are supposed to be "sequences" in YAML, I wonder why "flow style" ones don't work

Comment (rezbyte) [reply .github/workflows/kotlin-test.yml:7]: Yeah, it's weird. My only other hypothesis is the different quotation marks. YAML handles wildcards differently depending on quote style from memory. It still does not make much sense to me either way to be honest.

**Pull request merged**